### PR TITLE
Espnet3/060125

### PR DESCRIPTION
**Pull request by Masao-Someki**
## What?

Contains:
1. Refactoring recipes
2. Bug fix on Trainer (when using PyTorch's default dataloader with distributed training)
3. Bugfix / feature enhancements on inference/scoring stage

## Why?

espnet-3

### CODE REVIEW

**Comment by Masao-Someki** [comment]
I trained OWSM with 10% subset of data, and compared with espnet-2 exp. The performance gets better, one of the reason I guess is that during training I found in espnet-2's exp we had `NaN` value in loss/grad, while we didn't have them in espnet-3.

image

**Comment by Masao-Someki** [comment]
I will address the CI in the next PR, focusing on adding tests for inference and scoring, formatting and refactoring, and enhancing the documentation.
